exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -591,201 +889,382 @@ func (e *Exchange) CancelBatchOrders(ctx context.Context, o []order.Cancel) (*or
-			Address: address,
-			Tag:     tag,
-		}, nil
-	}
-
-	// Handle coins like BTC or multichain coins
-	targetCurrency := cryptocurrency.String()
-	if chain != "" && !strings.EqualFold(chain, cryptocurrency.String()) {
-		targetCurrency = chain
 	}
 
-	address, ok = depositAddrs.Addresses[strings.ToUpper(targetCurrency)]
+	var chainName string
+	address, ok := (*depositAddrs)[cryptocurrency.Upper().String()]
 	if !ok {
-		if len(coinParams.ChildChains) > 1 && chain != "" && !slices.Contains(coinParams.ChildChains, targetCurrency) {
-			// rather than assume, return an error
-			return nil, fmt.Errorf("currency %s has %v chains available, one of these must be specified",
-				cryptocurrency,
-				coinParams.ChildChains)
-		}
-
-		coinParams, ok = currencies[strings.ToUpper(targetCurrency)]
-		if !ok {
-			return nil, fmt.Errorf("unable to find currency %s in map", cryptocurrency)
-		}
-		if coinParams.WithdrawalDepositDisabled == 1 {
-			return nil, fmt.Errorf("deposits and withdrawals for %v are currently disabled", targetCurrency)
-		}
+		for _, networkDetail := range currencyDetail.NetworkList {
+			if !networkDetail.DepositEnable {
+				return nil, fmt.Errorf("deposits and withdrawals for %v are currently disabled", cryptocurrency.Upper().String())
+			}
 
-		newAddr, err := e.GenerateNewAddress(ctx, targetCurrency)
-		if err != nil {
-			return nil, err
+			newAddr, err := e.NewCurrencyDepositAddress(ctx, cryptocurrency)
+			if err != nil {
+				return nil, err
+			}
+			address = newAddr
+			chainName = networkDetail.Blockchain
 		}
-		address = newAddr
 	}
-	return &deposit.Address{Address: address}, nil
+	return &deposit.Address{
+		Address: address,
+		Chain:   chainName,
+	}, nil
 }
 
 // WithdrawCryptocurrencyFunds returns a withdrawal ID when a withdrawal is
 // submitted
 func (e *Exchange) WithdrawCryptocurrencyFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
-	if err := withdrawRequest.Validate(); err != nil {
-		return nil, err
+	if withdrawRequest == nil {
+		return nil, withdraw.ErrRequestCannotBeNil
 	}
-
-	targetCurrency := withdrawRequest.Currency.String()
-	if withdrawRequest.Crypto.Chain != "" {
-		targetCurrency = withdrawRequest.Crypto.Chain
+	if withdrawRequest.Amount <= 0 {
+		return nil, limits.ErrAmountBelowMin
 	}
-	v, err := e.Withdraw(ctx, targetCurrency, withdrawRequest.Crypto.Address, withdrawRequest.Amount)
+	if withdrawRequest.Currency.IsEmpty() {
+		return nil, currency.ErrCurrencyCodeEmpty
+	}
+	v, err := e.WithdrawCurrency(ctx, &WithdrawCurrencyRequest{
+		Coin:       withdrawRequest.Currency,
+		Network:    withdrawRequest.Crypto.Chain,
+		Address:    withdrawRequest.Crypto.Address,
+		Amount:     withdrawRequest.Amount,
+		AddressTag: withdrawRequest.Crypto.AddressTag,
+	})
 	if err != nil {
 		return nil, err
 	}
 	return &withdraw.ExchangeResponse{
-		Status: v.Response,
+		Name: e.Name,
+		ID:   strconv.FormatInt(v.WithdrawRequestID, 10),
```

Comment:
🍺 This looks like the field `WithdrawRequestID` might be a string

Reply:
<img width="597" height="199" alt="Screenshot 2025-10-23 at 9 42 46 in the morning" src="https://github.com/user-attachments/assets/a20f3ddb-e976-47bc-a6cd-934918b3e6ca" />